luarules/gadgets/unit_zombies.lua

```diff
@@ -0,0 +1,465 @@
+local ZOMBIES_REZ_SPEED = 16			--metal per second
+local ZOMBIES_COUNT_MIN = 1
+local ZOMBIES_COUNT_MAX = 1
+local ZOMBIES_REZ_MIN = 60				-- in seconds
+local ZOMBIE_REZ_MAX = 180				-- in seconds
+
+if modOptions.zombies == "hard" then
+	ZOMBIES_REZ_SPEED = 24
+	ZOMBIES_REZ_MIN = 30
+	ZOMBIE_REZ_MAX = 120
+elseif modOptions.zombies == "nightmare" then
+	ZOMBIES_REZ_SPEED = 24
+	ZOMBIES_COUNT_MIN = 2
+	ZOMBIES_COUNT_MAX = 5
+	ZOMBIES_REZ_MIN = 30
+	ZOMBIE_REZ_MAX = 120
+end
+
+local ZOMBIE_ORDER_CHECK_INTERVAL = Game.gameSpeed * 10 -- How often (in frames) to check if zombies need new orders
+local ZOMBIE_CHECK_INTERVAL = Game.gameSpeed -- How often (in frames) everything else is checked
+
+local CMD_REPEAT = CMD.REPEAT
+local CMD_MOVE_STATE = CMD.MOVE_STATE
+local CMD_FIGHT = CMD.FIGHT
+local CMD_OPT_SHIFT = CMD.OPT_SHIFT
+local CMD_GUARD = CMD.GUARD
+local CMD_FIRE_STATE = CMD.FIRE_STATE
+
+local ISNT_ZOMBIE = 0
+local IS_ZOMBIE = 1
+
+local spGetGroundHeight			= Spring.GetGroundHeight
+local spGetUnitPosition			= Spring.GetUnitPosition
+local spGetFeaturePosition		= Spring.GetFeaturePosition
+local spCreateUnit				= Spring.CreateUnit
+local spTransferUnit			= Spring.TransferUnit
+local spGetUnitDefID			= Spring.GetUnitDefID
+local spGetUnitTeam				= Spring.GetUnitTeam
+local spGetAllUnits				= Spring.GetAllUnits
+local spGetGameFrame			= Spring.GetGameFrame
+local spGetAllFeatures			= Spring.GetAllFeatures
+local spGiveOrderToUnit			= Spring.GiveOrderToUnit
+local spGetUnitCommandCount		= Spring.GetUnitCommandCount
+local spDestroyFeature			= Spring.DestroyFeature
+local spGetUnitIsDead			= Spring.GetUnitIsDead
+local spGiveOrderArrayToUnitArray = Spring.GiveOrderArrayToUnitArray
+local spGetUnitsInCylinder		= Spring.GetUnitsInCylinder
+local spSetTeamResource			= Spring.SetTeamResource
+local spGetUnitHealth			= Spring.GetUnitHealth
+local spSetUnitHealth			= Spring.SetUnitHealth
+local spSetUnitRulesParam		= Spring.SetUnitRulesParam
+local spGetUnitRulesParam		= Spring.GetUnitRulesParam
+local spGetFeatureDefID			= Spring.GetFeatureDefID
+local spTestMoveOrder			= Spring.TestMoveOrder
+local spSpawnCEG				= Spring.SpawnCEG
+local spGetFeatureResources		= Spring.GetFeatureResources
+local spGetFeatureHealth		= Spring.GetFeatureHealth
+local spDestroyUnit				= Spring.DestroyUnit
+local spCreateFeature			= Spring.CreateFeature
+local spSpawnExplosion			= Spring.SpawnExplosion
+local spPlaySoundFile			= Spring.PlaySoundFile
+local spGetFeatureRadius		= Spring.GetFeatureRadius
+local spGetUnitCurrentCommand	= Spring.GetUnitCurrentCommand
+local spSetUnitExperience		= Spring.SetUnitExperience
+local random					= math.random
+local distance2dSquared			= math.distance2dSquared
+
+local teams = Spring.GetTeamList()
+local scavTeamID
+local gaiaTeamID = Spring.GetGaiaTeamID()
+for _, teamID in ipairs(teams) do
+
+	local teamLuaAI = Spring.GetTeamLuaAI(teamID)
+	if (teamLuaAI and string.find(teamLuaAI, "ScavengersAI")) then
+		scavTeamID = teamID
```

Comment:
@WatchTheFort is there some sort of `Spring.Utilities.GetScavTeamID()` helper? Eventually there should be one (but if there isn't then it's fine as-is, adding a helper is not the scope of this patch).

Reply:
Not yet, but we definitely need to add it